File: acme-java-reactive/src/main/java/acme/repository/CustomerAccountRepository.java
package acme.repository;

import acme.entity.CustomerAccount;
import org.springframework.data.repository.reactive.ReactiveCrudRepository;
import org.springframework.stereotype.Repository;
import reactor.core.publisher.Flux;

import java.util.UUID;

@Repository
public interface CustomerAccountRepository extends ReactiveCrudRepository<CustomerAccount, UUID> {
    Flux<CustomerAccount> findByCustomerId(UUID customerId);
}

File: acme-springboot-nonblocking/src/main/kotlin/acme/repository/CustomerAccountRepository.kt
package acme.repository

import acme.entity.CustomerAccount
import kotlinx.coroutines.flow.Flow
import org.springframework.data.repository.kotlin.CoroutineCrudRepository
import org.springframework.stereotype.Repository
import java.util.*

@Repository
interface CustomerAccountRepository : CoroutineCrudRepository<CustomerAccount, UUID> {
    fun findByCustomerId(customerId: UUID): Flow<CustomerAccount>
}
